src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,115 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.NoSuchElementException;
+
 public class WorkWithFile {
+    private static final String SPLITERATOR = ",";
+    private static final String CASE_BUY = "buy";
+    private static final String CASE_SUPPLY = "supply";
+    private static final String RESULT = "result";
+    private static final String EXCEPTION_MESSAGE = "Key doesn't found";
+    private static final String PARSE_EXCEPTION_MESSAGE = "Something went wrong";
+    private static final int SUPPLY_VALUE = 1;
+    private static final int BUY_VALUE = 3;
+    private static final int KEY = 0;
+    private static final int KEY_VALUE = 1;
+    private static final String SPLIT_BY_NEW_LINE = System.lineSeparator();
+    private static final int END_FILE_CODE = -1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int sumForBuy = 0;
+        int sumForSupply = 0;
+        StringBuilder textForSavingToFile = new StringBuilder();
+        StringBuilder constructTextFromFile = new StringBuilder();
+        try (
+                BufferedReader reader = new BufferedReader(new FileReader(fromFileName));
+                FileWriter writer = new FileWriter(toFileName)
+        ) {
+            readFile(reader, constructTextFromFile);
+            String[] splitConstructTextFromFile =
+                    getSplitConstructTextFromFile(constructTextFromFile);
+
+            String[] processData = processData(splitConstructTextFromFile, sumForSupply, sumForBuy);
+
+            StringBuilder textForSavingFile =
+                    generateReport(
+                            Integer.parseInt(processData[BUY_VALUE]),
+                            Integer.parseInt(processData[SUPPLY_VALUE]),textForSavingToFile);
+            writeToFile(textForSavingFile.toString(), writer);
+
+        } catch (IOException e) {
+            throw new NoSuchElementException(PARSE_EXCEPTION_MESSAGE);
```

Comment:
![image](https://github.com/mate-academy/jv-work-with-file/assets/63095184/88d2ba75-6d1c-4873-b34e-b1a11ff00309)
Use this Structure in this method